agent/protocol/nats/nats.go

```diff
@@ -0,0 +1,771 @@
+	}
+	common.ProtocolParserLog.Debugf("NATS Parsed Unsub:[%v], ReadBytes:%d", msg.String(), prasedLen)
+	return &msg.NatsMessage, prasedLen
+}
+
+// Msg
+func (m *Msg) String() string {
+	return fmt.Sprintf("Protocol:MSG,Subject:%v,Sid:%v,ReplyTo:%v,PayloadSize:%d,Payload:%x",
+		m.Subject, m.Sid, m.ReplyTo, m.PayloadSize, m.Payload)
+}
+
+func (m *Msg) ParseData(payload []byte) (*Msg, int) {
+	// MSG <subject> <sid> [reply-to] <#bytes>␍␊[payload]␍␊
+	packetLen := readLine(payload)
+	if packetLen < 0 {
+		return nil, -1
+	}
+	parts := splitFields(payload[:packetLen-len(_CRLF_)])
+	if len(parts) < 4 || len(parts) > 5 {
+		return nil, packetLen
+	}
+	// if strings.ToUpper(string(parts[0])) != "MSG" {
+	// 	return nil, packetLen
+	// }
+
+	msg := Msg{}
+	msg.Subject = string(parts[1])
+	msg.Sid = string(parts[2])
+
+	index := 3
+	if len(parts) == 5 {
+		msg.ReplyTo = string(parts[3])
+		index++
+	}
+
+	payloadSize, err := strconv.Atoi(string(parts[index]))
+	if err != nil {
+		return nil, packetLen
+	}
+	packetEnd := packetLen + payloadSize + len(_CRLF_)
+	if len(payload) < packetEnd {
+		return nil, packetLen
+	}
+	if !bytes.HasPrefix(payload[packetLen+payloadSize:], []byte(_CRLF_)) {
+		return nil, packetLen
+	}
+	msg.PayloadSize = payloadSize
+	msg.Payload = payload[packetLen : packetLen+payloadSize]
+
+	msg.ProtocolCode = MSG
+	msg.isReq = false
+	msg.NatsMessage.Subject = msg.Subject
+	msg.Buf = payload[:packetEnd]
+	return &msg, packetEnd
+}
+
+func (m *Msg) Parse(payload []byte, messageType protocol.MessageType) (*NatsMessage, int) {
+	common.ProtocolParserLog.Debugf("NATS Parse Msg:%d, %x", len(payload), string(payload))
+	msg, prasedLen := m.ParseData(payload)
+	if prasedLen < 0 || msg == nil {
+		return nil, prasedLen
+	}
+	common.ProtocolParserLog.Debugf("NATS Parsed Msg:[%v], ReadBytes:%d", msg.String(), prasedLen)
+	return &msg.NatsMessage, prasedLen
+}
+
+// Hmsg
+func (m *Hmsg) String() string {
+	return fmt.Sprintf("Protocol:HMSG,Subject:%v,Sid:%v,ReplyTo:%v,HeaderSize:%d,PayloadSize:%d,HeaderVersion:%v,Headers:%v,Payload:%x",
+		m.Subject, m.Sid, m.ReplyTo, m.HeaderSize, m.PayloadSize, m.HeaderVersion, m.Headers, m.Payload)
+}
+
+func (m *Hmsg) ParseData(payload []byte) (*Hmsg, int) {
+	// HMSG <subject> <sid> [reply-to] <#header bytes> <#total bytes>␍␊[headers]␍␊␍␊[payload]␍␊
+	packetLen := readLine(payload)
+	if packetLen < 0 {
+		return nil, -1
+	}
+	parts := splitFields(payload[:packetLen-len(_CRLF_)])
+	if len(parts) < 5 || len(parts) > 6 {
+		return nil, packetLen
+	}
+	// if strings.ToUpper(string(parts[0])) != "HMSG" {
+	// 	return nil, packetLen
+	// }
+
+	msg := Hmsg{}
+	msg.Subject = string(parts[1])
+	msg.Sid = string(parts[2])
+
+	index := 3
+	if len(parts) == 6 {
+		msg.ReplyTo = string(parts[3])
+		index++
+	}
+
+	headerSize, err_1 := strconv.Atoi(string(parts[index]))
+	totalSize, err_2 := strconv.Atoi(string(parts[index+1]))
+	if err_1 != nil || err_2 != nil || headerSize > totalSize {
+		return nil, packetLen
+	}
```

Comment:
That's not idiomatic

Try this


```suggestion
	headerSize, err := strconv.Atoi(string(parts[index]))
	if err != nil {
		return nil, packetLen
	}

	totalSize, err := strconv.Atoi(string(parts[index+1]))
	if err != nil || headerSize > totalSize {
		return nil, packetLen
	}
```